### PR TITLE
feat(validate_address): add validate address rpc

### PR DESCRIPTION
**Pull request by Milerius**
To test:

- [x] invalid bch address
- [x] legacy bch address
- [x] BCH address with bitcoincash: prefix
- [x] BCH address without bitcoincash: prefix
- [x] invalid eth address
- [x] lower case valid eth address
- [x] send rick/morty/utxo in general
- [x] try specific BEP20 / ERC20 token withdraw

### CODE REVIEW

File: atomic_defi_design/qml/Wallet/SendModal.qml [hunk 1]
```diff
@@ -239,7 +259,7 @@ BasicModal {
         // ERC-20 Lowercase issue
         RowLayout {
             Layout.fillWidth: true
-            visible: isERC20() && input_address.field.text != "" && hasErc20CaseIssue(input_address.field.text)
+            visible: needFix//isERC20() && input_address.field.text != "" && hasErc20CaseIssue(input_address.field.text)
```
Comment (Milerius) [inline atomic_defi_design/qml/Wallet/SendModal.qml:262]: better to call it `convertible`

File: atomic_defi_design/qml/Wallet/SendModal.qml [hunk 2]
```diff
@@ -249,7 +269,9 @@ BasicModal {
             DefaultButton {
                 Layout.alignment: Qt.AlignRight
                 text: qsTr("Fix")
-                onClicked: input_address.field.text = API.app.to_eth_checksum_qt(input_address.field.text.toLowerCase())
+                onClicked: {
+                    api_wallet_page.convert_address(input_address.field.text, address_data.to_address_format)
```
Comment (Milerius) [inline atomic_defi_design/qml/Wallet/SendModal.qml:273]: This is good but we still need to listen to the update of the converted_address string and assign it to the input text once is changing

**Comment by smk762** [comment]
- lowercase ETH shows "Fix button", and converts to mix case that allows tx preparation.
- if ETH address length incorrect, warning displayed.
- BCH q address without `bitcoincash:` prefix ok
- with prefix, error - `Invalid base32 input string`, does not accept `:` as input.
- legacy address reveals "fix" button, which converts to new format (with prefix) as expected.
- RICK/MORTY transactions ok
- invalid BCH address shows warning (checksum fail)

**Comment by Milerius** [comment]
> * lowercase ETH shows "Fix button", and converts to mix case that allows tx preparation.
> * if ETH address length incorrect, warning displayed.
> * BCH q address without `bitcoincash:` prefix ok
> * with prefix, error - `Invalid base32 input string`, does not accept `:` as input.
> * legacy address reveals "fix" button, which converts to new format (with prefix) as expected.
> * RICK/MORTY transactions ok
> * invalid BCH address shows warning (checksum fail)

@ssakone can you please allow the pasting of bitcoincash address with prefix in the front end ?

**Comment by Milerius** [comment]
@smk762 you can test again i think you should be able to paste `:` now

File: atomic_defi_design/qml/Components/AddressFieldWithTitle.qml [hunk 3]
```diff
@@ -3,10 +3,9 @@ import QtQuick 2.15
...
 
-    field.validator: RegExpValidator {
-        regExp: /[a-zA-Z0-9 \t]{25,50}/
-    }
+
     field.onTextChanged: {
+        field.text = field.text.trim()
         if(field.text.indexOf(' ') !== -1 || field.text.indexOf('\t') !== -1) {
             field.text = field.text.replace(/[ \t]/, '')
         }
```
Comment (Milerius) [inline atomic_defi_design/qml/Components/AddressFieldWithTitle.qml:11]: This is probably not required anymore @ssakone since you trim before

**Review by smk762 — APPROVED** [review]
bch prefix working now, thanks.

**Pull request merged**